server/games/game_results.py

```diff
@@ -75,21 +75,37 @@ def outcome(self, army: int) -> GameOutcome:
         if army not in self:
             return GameOutcome.UNKNOWN
 
-        outcomes = set(
-            r.outcome for r in self[army]
-            if r.outcome is not GameOutcome.UNKNOWN
+        voters = defaultdict(set)
+        for report in filter(
+            lambda r: r.outcome is not GameOutcome.UNKNOWN, self[army]
+        ):
+            voters[report.outcome].add(report.reporter)
+
+        if len(voters) == 0:
+            return GameOutcome.UNKNOWN
+
+        if len(voters) == 1:
+            unique_outcome = next(iter(voters.keys()))
+            return unique_outcome
+
+        sorted_outcomes = sorted(
+            list(voters.keys()),
+            reverse=True,
+            key=lambda x: (len(voters[x]), x.value)
         )
 
-        if len(outcomes) == 1:
-            return outcomes.pop()
-        elif len(outcomes) > 1:
-            self._logger.info(
-                "Multiple outcomes for game %s army %s: %s",
-                self._game_id, army, list(outcomes)
-            )
-            return GameOutcome.CONFLICTING
+        top_votes = len(voters[sorted_outcomes[0]])
+        runnerup_votes = len(voters[sorted_outcomes[1]])
+        if top_votes > 1 >= runnerup_votes or top_votes >= runnerup_votes + 3:
+            decision = sorted_outcomes[0]
+        else:
+            decision = GameOutcome.CONFLICTING
 
-        return GameOutcome.UNKNOWN
+        self._logger.info(
+            f"Multiple outcomes for game {self._game_id} army {army} "
```

Comment:
Not that it matters here, but its better to use `%s` and let the logger do string formatting lazily. 

If you do it that way all the time, then you will never accidentally shoot yourself in the foot when the performance actually matters :P

Reply:
Ah, I wasn't really aware. I just went with the more readable one.